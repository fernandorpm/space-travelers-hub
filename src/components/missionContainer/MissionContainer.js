import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
// import { motion } from 'framer-motion';
import { fetchAllMissions } from '../../redux/missions/missionsSlice';
import Mission from '../mission/Mission';
import styles from './missionContainerStyle.module.scss';

const MissionContainer = () => {
  const dispatch = useDispatch();
  const allMissions = useSelector((state) => state.missions.entities);

  useEffect(() => {
    dispatch(fetchAllMissions());
  }, [dispatch]);

  const createMissions = (list) => list.map((mission) => (
    <Mission
      key={mission.mission_id}
      id={mission.mission_id}
      name={mission.mission_name}
      description={mission.description}
      member={mission.reserved}
    />
  ));

  return (
    <>
      {allMissions
      && (
      <table className={styles.mainContainer}>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
        {createMissions(allMissions)}
      </table>
      )}
    </>
  );
};

export default MissionContainer;
