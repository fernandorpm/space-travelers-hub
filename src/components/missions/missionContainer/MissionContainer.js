import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { motion } from 'framer-motion';
import { fetchAllMissions } from '../../../redux/missions/missionsSlice';
import Mission from '../mission/Mission';
import styles from './missionContainerStyle.module.scss';

const container = {
  initial: {},
  animate: {
    transition: {
      staggerChildren: 0.1,
      delayChildren: 0.3,
    },
  },
};

const MissionContainer = () => {
  const dispatch = useDispatch();
  const allMissions = useSelector((state) => state.missions.entities);

  useEffect(() => {
    if (allMissions.length === 0) {
      dispatch(fetchAllMissions());
    }
  }, [dispatch]);

  const createMissions = (list) => list.map((mission) => (
    <Mission
      key={mission.mission_id}
      id={mission.mission_id}
      name={mission.mission_name}
      description={mission.description}
      joined={mission.reserved}
    />
  ));

  return (
    <>
      {allMissions
      && (
      <motion.table
        className={styles.mainContainer}
        variants={container}
        initial="initial"
        animate="animate"
      >
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
        {createMissions(allMissions)}
      </motion.table>
      )}
    </>
  );
};

export default MissionContainer;
