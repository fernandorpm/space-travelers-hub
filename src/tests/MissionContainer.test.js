import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MissionContainer from '../components/missions/missionContainer/MissionContainer';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><MissionContainer /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
