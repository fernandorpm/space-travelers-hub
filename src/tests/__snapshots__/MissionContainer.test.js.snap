// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="mainContainer"
  style={Object {}}
>
  <tr>
    <th>
      Mission
    </th>
    <th>
      Description
    </th>
    <th>
      Status
    </th>
    <th>
       
    </th>
  </tr>
</table>
`;
